src/js_native_api_types.h

```diff
@@ -115,6 +142,13 @@ typedef napi_value(NAPI_CDECL* napi_callback)(napi_env env,
 typedef void(NAPI_CDECL* napi_finalize)(napi_env env,
                                         void* finalize_data,
                                         void* finalize_hint);
+#if defined(NAPI_EXPERIMENTAL) && defined(NODE_API_EXPERIMENTAL_NOGC_ENV)
+typedef void(NAPI_CDECL* node_api_nogc_finalize)(node_api_nogc_env env,
```

Comment:
Can we keep a single finalize callback? Are there situations where we must use one vs another?

Reply:
As we discussed in today's meeting, we need the second type of callback in order to pass functions of that type to `node_api_post_finalizer`, because a callback passed to this new API must be identical to a regular `napi_finalizer`, but must have access to all Node-APIs.